src/nvim/eval.c

```diff
@@ -16801,17 +16815,17 @@ static void f_type(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   int n = -1;
 
   switch (argvars[0].v_type) {
-    case VAR_NUMBER: n = 0; break;
-    case VAR_STRING: n = 1; break;
-    case VAR_FUNC:   n = 2; break;
-    case VAR_LIST:   n = 3; break;
-    case VAR_DICT:   n = 4; break;
-    case VAR_FLOAT:  n = 5; break;
+    case VAR_NUMBER: n = VAR_TYPE_NUMBER; break;
+    case VAR_STRING: n = VAR_TYPE_STRING; break;
+    case VAR_FUNC:   n = VAR_TYPE_FUNC; break;
+    case VAR_LIST:   n = VAR_TYPE_LIST; break;
+    case VAR_DICT:   n = VAR_TYPE_DICT; break;
+    case VAR_FLOAT:  n = VAR_TYPE_FLOAT; break;
     case VAR_SPECIAL: {
       switch (argvars[0].vval.v_special) {
         case kSpecialVarTrue:
         case kSpecialVarFalse: {
-          n = 6;
+          n = VAR_TYPE_BOOL;
           break;
         }
         case kSpecialVarNull: {
```

Comment:
Okay, yes, it is kind of silly because even the vim help itself on `type()` suggests a way to not use magic numbers (e.g. `type('[]')` for seeing if something is a list. Let's let this PR languish.